common/src/main/scala/common/util/IORetry.scala

```diff
@@ -33,23 +33,23 @@ object IORetry {
                       state: S,
                       maxRetries: Option[Int],
                       backoff: Backoff,
-                      isTransient: Throwable => Boolean = throwableToFalse,
-                      isFatal: Throwable => Boolean = throwableToFalse,
+                      isRetryable: Throwable => Boolean = throwableToTrue,
```

Comment:
TOL it's a little confusing that we have two functions to return almost the same information here. And because they're separate, it's possible that they combine to produce an invalid state. Maybe a `Retryability` category case class with three options could more precisely capture the designation here?

```scala
sealed trait IoErrorRetryabilityCategory
case object FatalIoError extends IoRetryabilityCategory
case object RetryableIoError extends IoRetryabilityCategory
case object InfinitelyRetryableIoError extends IoRetryabilityCategory
```

Then you could have a single input function like:
```scala
retryabilityCategorizationFunction: Throwable => IoErrorRetryabilityCategory
```